sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -17,22 +56,299 @@ class TableCheckpointStore:
+        :keyword str api_version:
+            The Storage API version to use for requests. Default value is '2018-03-28'.
+        :returns: A table checkpoint store.
+        :rtype: ~azure.eventhub.extensions.checkpointstoretable.TableCheckpointStore
+        """
+        endpoint, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, keyword_args=kwargs
+        )
+        return cls(endpoint, table_name=table_name, credential=credential, **kwargs)
 
-    def list_ownership(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def __enter__(self):
+        self._table_client.__enter__()
+        return self
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
+    def __exit__(self, *args):
+        self._table_client.__exit__(*args)
+
+    @classmethod
+    def _create_ownership_entity(cls, ownership):
+        """
+        Create a dictionary with the `ownership` attributes.
+        """
+        ownership_entity = {
+            "PartitionKey": "{} {} {} Ownership".format(
+                ownership["fully_qualified_namespace"],
+                ownership["eventhub_name"],
+                ownership["consumer_group"],
+            ),
+            "RowKey": ownership["partition_id"],
+            "ownerid": ownership["owner_id"],
+        }
+        return ownership_entity
+
+    @classmethod
+    def _create_checkpoint_entity(cls, checkpoint):
+        """
+        Create a dictionary with `checkpoint` attributes.
+        """
+        checkpoint_entity = {
+            "PartitionKey": "{} {} {} Checkpoint".format(
+                checkpoint["fully_qualified_namespace"],
+                checkpoint["eventhub_name"],
+                checkpoint["consumer_group"],
+            ),
+            "RowKey": checkpoint["partition_id"],
+            "offset": checkpoint["offset"],
+            "sequencenumber": checkpoint["sequence_number"],
+        }
+        return checkpoint_entity
+
+    def _update_ownership(self, ownership, **kwargs):
+        """_update_ownership mutates the passed in ownership."""
+        try:
+            ownership_entity = TableCheckpointStore._create_ownership_entity(ownership)
+            metadata = self._table_client.update_entity(
+                mode=UpdateMode.REPLACE,
+                entity=ownership_entity,
+                etag=ownership["etag"],
+                match_condition=MatchConditions.IfNotModified,
+                **kwargs
+            )
+            ownership["etag"] = metadata["etag"]
+            updated_entity = self._table_client.get_entity(
+                partition_key=ownership_entity["PartitionKey"],
+                row_key=ownership_entity["RowKey"],
+                **kwargs
+            )
+            ownership["last_modified_time"] = _to_timestamp(
+                updated_entity.metadata.get("timestamp")
+            )
+            return ownership
```

Comment:
since this method mutates the original ownership that was passed in, you don't need to return it again (esp since the returned value is not being used by the method that calls this one)
```suggestion
```